src/server/cluster/incoming_slot_migration.h

```diff
@@ -34,8 +34,12 @@ class IncomingSlotMigration {
   // Stop and join the migration, can be called even after migration is finished
   void Stop();
 
+  // Init/Reinit migration
+  void Init(uint32_t shards_num);
+
   MigrationState GetState() const {
-    return state_.load();
+    util::fb2::LockGuard lk(state_mu_);
```

Comment:
Is there a reason we use a lock here? What is the reason we replaced the atomic variable with the mutex ?